### PR TITLE
chore: loads file inside operator when using FromFile.

### PR DESCRIPTION
**Pull request by jcchavezs**
Currently files are loaded on parsing and passed down to operators (e.g. pmFromFile) inside the options, this is inconvenient because we have custom logic for them (making useless the interface) but also making it hard to implement things like lazy load of the data.

Ping @anuraaga 

> Thank you for contributing to Coraza WAF, your effort is greatly appreciated
> Before submitting check if what you want to add to `coraza` list meets [quality standards](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#quality-standards) before sending pull request. Thanks!

**Note**: _that go.mod and go.sum can only be modified for tested dependency updates or justified new features._

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v2sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).

Thanks for your PR :heart:

### CODE REVIEW

File: seclang/rule_parser.go [hunk 1]
```diff
@@ -194,22 +192,13 @@ func (p *RuleParser) ParseOperator(operator string) error {
...
-			return err
-		}
-		opdata = tpath
-		data = content
-	}
+
 	opts := coraza.RuleOperatorOptions{
 		Arguments: string(data),
+		Path: []string{
+			p.options.Config.Get("parser_config_dir", "").(string),
```
Comment (jcchavezs) [inline seclang/rule_parser.go:199]: I am thinking maybe we need a `Config.GetString` function to avoid casting.

Comment (jptosso) [reply seclang/rule_parser.go:199]: Go best practices state that we should use Config.String().

I don't see any problem with casting, though, if it fails, Get will fall back to the default value. You can also use type assertion. Feel free to write the PR though

**Review by jptosso — APPROVED** [review]
LGTM

File: operators/from_file.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+package operators
+
+import (
+	"os"
+	"path"
+)
+
+func loadFromFile(filepath string, paths []string) ([]byte, error) {
```
Comment (anuraaga) [inline operators/from_file.go:11]: Might consider accepting `[]fs.FS` instead of strings for paths, this would allow someone to use an `go:embed` file system as roots if they needed to.

File: operators/from_file.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	if path.IsAbs(filepath) {
+		return os.ReadFile(filepath)
+	}
+
+	// handling files by operators is hard because we must know the paths where we can
+	// search, for example, the policy path or the binary path...
+	// CRS stores the .data files in the same directory as the directives
+	var (
```
Comment (anuraaga) [inline operators/from_file.go:19]: Not sure if this is needed

File: operators/from_file_test.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	"path/filepath"
+	"testing"
+)
+
+const fileContent = "abc123"
+
+func getTestFile(t *testing.T) (string, string) {
+	t.Helper()
+	tmpDir := t.TempDir()
+	tmpFile, err := os.Create(filepath.Join(tmpDir, "tmpfile"))
```
Comment (anuraaga) [inline operators/from_file_test.go:17]: Could be a bit shorter using `os.WriteFile` I think

File: operators/from_file.go (hunk 3 above)
Comment (jcchavezs) [reply operators/from_file.go:19]: The var or the comment? if it is the var I used it to declare error on beforehand to be able to return it in the last part.

File: operators/from_file.go (hunk 2 above)
Comment (jcchavezs) [reply operators/from_file.go:11]: Not sure I understand the use case. This is more like an internal function that is used by the `*FromFile` functions, I don't think people might call that directly.

File: operators/from_file.go (hunk 3 above)
Comment (anuraaga) [reply operators/from_file.go:19]: Oh I didn't notice the `err` being returned. It looks like if there are no paths, `err` won't be returned, so maybe you should return `os.ErrNotExist` explicitly there? Either way it could be clearer, predeclaring here seems to be making it hard to follow what makes it into the variable by the end of the function
